File: aifo/slideinsight/internal/server/routes/permissions/permissions.go
// Copyright 2025 Jonas Teuwen. All rights reserved.
//
// This file is part of SlideInsight.
//
// Use of this source code is governed by the terms found in the
// LICENSE file located in the SlideInsight project root.
package permissions

import (
	"aifo.dev/aifo/slideinsight/internal/config"
	"aifo.dev/aifo/slideinsight/internal/server/handlers"
	"aifo.dev/aifo/slideinsight/internal/server/middleware"
	"aifo.dev/aifo/slideinsight/internal/server/services"

	"github.com/gofiber/fiber/v2"
)

// SetupPermissionRoutes configures all permission related routes
func SetupPermissionRoutes(apiRoutes fiber.Router, permissionService services.PermissionService, config config.Config) {
	// Permission routes group
	permissionRoutes := apiRoutes.Group("/v1/permissions")

	// All permission routes require authentication
	permissionRoutes.Use(middleware.Protected(config.Auth))

	// Permission management routes
	permissionRoutes.Post("/", handlers.CreatePermission(permissionService))
	permissionRoutes.Post("/bulk", handlers.CreatePermissionsBulk(permissionService))
	permissionRoutes.Get("/", handlers.GetPermissions(permissionService))
	permissionRoutes.Get("/:name", handlers.GetPermissionByName(permissionService))
	permissionRoutes.Delete("/:name", handlers.DeletePermission(permissionService))
}

File: aifo/slideinsight/ui/src/types/permissions.ts
// Copyright 2025 Jonas Teuwen. All rights reserved.
//
// This file is part of SlideInsight.
//
// Use of this source code is governed by the terms found in the
// LICENSE file located in the SlideInsight project root.

export interface Permission {
  id: number;
  tenant_id: number; // 0 = system tenant, >0 = regular tenant
  short_uid: string;
  name: string;
  description: string;
  created_at: string;
  updated_at: string;
}

export interface CreatePermissionRequest {
  name: string;
  description: string;
}

// Permission filter parameters for API requests
export interface PermissionFilterParams {
  filterAccessibleStudies?: boolean;
  filterAccessibleCases?: boolean;
  filterAccessibleSlides?: boolean;
}

export interface PermissionsResponse {
  permissions?: Permission[];
}

interface CreatePermissionResponse {
  permission: Permission;
}
